encryption-pvc-kms-vault-token-auth test is failing

[encryption-pvc-kms-vault-token-auth](https://github.com/rook/rook/runs/6160644379?check_suite_focus=true#logs) canary test is failing in `deploy vault` stage. 

error
```
+ kubectl exec -ti vault-0 -- vault operator init -format json -ca-cert /vault/userconfig/vault-server-tls/vault.crt
Unable to use a TTY - input is not a terminal or the right kind of file
error: unable to upgrade connection: container not found ("vault")
Error: Process completed with exit code 1.
```

--- a/tests/scripts/generate-tls-config.sh
+++ b/tests/scripts/generate-tls-config.sh
@@ -74,6 +74,13 @@ kubectl create -f "${DIR}/"csr.yaml
 
 kubectl certificate approve "${CSR_NAME}"
 
+timeout 10 bash <<-'EOF'
+    until [ $(kubectl get csr "${CSR_NAME}" -o jsonpath='{.status.certificate}' | wc -c) -gt 1  ]; do
+      echo "waiting for certificate "${CSR_NAME}" to be filled"
+      sleep 1
+    done
+EOF
+
 serverCert=$(kubectl get csr "${CSR_NAME}" -o jsonpath='{.status.certificate}')
 echo "${serverCert}" | openssl base64 -d -A -out "${DIR}"/"${SERVICE}".crt
 kubectl config view --raw --minify --flatten -o jsonpath='{.clusters[].cluster.certificate-authority-data}' | base64 -d > "${DIR}"/"${SERVICE}".ca

